### PR TITLE
DOC: Cleanup wav read/write docs and add tables for common types

### PR DESCRIPTION
**Pull request by endolith**

### CODE REVIEW

**Comment by endolith** [comment]
also I think it's a bug that `write` accepts int8 type?  should only accept uint8?

File: scipy/io/wavfile.py [hunk 1]
```diff
@@ -150,19 +151,19 @@ def _read_riff_chunk(fid):
...
     """
+    Open a WAV file
+
     Return the sample rate (in samples/sec) and data from a WAV file.
 
     Parameters
     ----------
     filename : string or open file handle
-        Input wav file.
+        Input wav file.  This can be an open file or a filename.
```
Comment (Eric89GXL) [inline scipy/io/wavfile.py]: I don't think we need this since it's in the types above?

File: scipy/io/wavfile.py [hunk 2]
```diff
@@ -170,17 +171,27 @@ def read(filename, mmap=False):
...
+    =====================  ===========  ===========  =============
+         WAV format            Min          Max       NumPy dtype
+    =====================  ===========  ===========  =============
+    32-bit floating-point  -1.0         +1.0         float32
+    32-bit PCM             -2147483648  +2147483647  int32
+    16-bit PCM             -32768       +32767       int16
+    8-bit PCM              0            255          uint8
+    =====================  ===========  ===========  =============
+
+    Note that 8-bit PCM is unsigned.
```
Comment (Eric89GXL) [inline scipy/io/wavfile.py]: Is there a reference or link you could add for this, like Wikipedia?

File: scipy/io/wavfile.py (hunk 1 above)
Comment (endolith) [inline scipy/io/wavfile.py]: Yeah, I just moved it from the Notes section

File: scipy/io/wavfile.py (hunk 2 above)
Comment (endolith) [inline scipy/io/wavfile.py]: > PCM data is two's-complement except for resolutions of 1-8 bits, which are represented as offset binary.

http://www-mmsp.ece.mcgill.ca/documents/audioformats/wave/wave.html

Comment (Eric89GXL) [inline scipy/io/wavfile.py]: Yeah that's probably not reliable enough to put in there. Good to know, though.

Comment (Eric89GXL) [inline scipy/io/wavfile.py]: (reliable enough a URL I mean)

File: scipy/io/wavfile.py (hunk 1 above)
Comment (Eric89GXL) [inline scipy/io/wavfile.py]: I don't think we really need it either place -- seems clear enough from the enumerated types. WDYT?

**Comment by endolith** [comment]
Ok, changed those things and rebased

**Comment by Eric89GXL** [comment]
Thanks @endolith

**Pull request merged**